lib/kv_storage_redis.ex

```diff
@@ -1,18 +1,116 @@
 defmodule KvStorageRedis do
   @moduledoc """
-  Documentation for `KvStorageRedis`.
+  Модуль для работы с БД Redis
   """
 
+  use GenServer
+
+  defstruct host: nil, port: nil, db: nil, conn: nil
+
+  ### Interface
+
+  @doc ~S"""
+  Запуск процесса
+
+  ## Примеры
+
+      iex> {:ok, _pid} = KvStorageRedis.start()
+
+  """
+  @spec start(any, any, any) :: :ignore | {:error, any} | {:ok, pid}
+  def start(host \\ "127.0.0.1", port \\ 6379, db \\ 0) do
+    GenServer.start_link(__MODULE__, %__MODULE__{host: host, port: port, db: db})
+  end
+
+  @doc ~S"""
+  Возвращает конфигурацию процесса
+
+  ## Примеры
+
+      iex> {:ok, pid} = KvStorageRedis.start()
+      iex> %KvStorageRedis{} = KvStorageRedis.config(pid)
+
+  """
+  @spec config(atom | pid | {atom, any} | {:via, atom, any}) :: any
```

Comment:
Первый параметр отправится напрямик в [`GenServer.call/3`](https://hexdocs.pm/elixir/GenServer.html#call/3), тип должен совпадать с https://hexdocs.pm/elixir/GenServer.html#t:server/0. Возвращаемый тип тоже, вроде, известен :)
```suggestion
  @spec config(GenServer.server()) :: t() # or %__MODULE__{} если тип не объявлен явно
```